index.coffee

```diff
@@ -1,60 +1,172 @@
-through = require 'through'
+stream = require 'readable-stream'
 
 module.exports = (requestModule, requestOpts, opts) ->
-	res = new Resumable(requestModule, requestOpts, opts)
-	return res.resume()
-
-class Resumable
-	constructor: (requestModule, requestOpts, opts) ->
-		@stream = through (data) =>
-			@bytesRead += data.length
-			@stream.emit('data', data)
+	return new ResumableRequest(requestModule, requestOpts, opts)
+
+module.exports.defaults = (defaults = {}) ->
+	(requestModule, requestOpts, opts) ->
+		opts = Object.assign {}, defaults, opts
+		return new ResumableRequest(requestModule, requestOpts, opts)
+
+cloneResponse = (response) ->
+	# creates an `http.IncomingMessage`-like stream, that can also be piped to.
+	s = new stream.PassThrough()
+	[ 'headers', 'httpVersion', 'method', 'rawHeaders', 'statusCode', 'statusMessage' ].forEach (prop) ->
+		s[prop] = response[prop]
+	return s
+
+throttle = (fn, interval) ->
```

Comment:
I wasn't aware of it and needed something equivalent to `.cancel` and `.apply` and didn't bother to search the internet because it's simple enough. I'll see how installing single lodash function works and update.